### PR TITLE
Use block indentation not token alignment rule for multiline lambdas

### LINKED ISSUE #3092: VB prettylister gets indent wrong for "End Sub" of a multiline lambda whose final line is a comment
In VB, if I have a multiline lambda whose last line is a comment, then it gets the spacing of the final "End Sub" incorrect.

REPRO:
vb-prettylist

Look at the above code. Try typing in a space or two or three spaces in front of either "End Sub" line, and then cursor off.

WHAT I EXPECT: the End Sub should snap to directly underneath the "Sub".
WHAT I GET: the End Sub snaps back to the start of the line.

Note that this problem doesn't repro if you put in enough spaces to have the End Sub directly line up under the Sub. In that case the End Sub stays exactly where you put it.

This problem also doesn't repro if the final line of the sub isn't a multiline comment.

(I haven't tried to see what happens with empty lines).

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #3092 

There are two different ways to get token/lines to align with each other in a formatting rule.  One is to use the align tokens operation and the other is to use the block indentation operation.  Unfortunately, the align token operation is not property plumbed all the way down to the trivia formatters that get used when there are intervening comments, so it gets ignored and ends up aligning against the indentation rules. Ideally, this would get fixed so that the alignment operation is honored.  However, this problem can also be fixed by switching to using a block indentation operation like xml literal blocks use.

@heejaechang @Pilchie please review

### CODE REVIEW

**Comment by Pilchie** [comment]
Can we add a test to cover this?

**Comment by mattwar** [comment]
Apparently there are already lots of test associated with this behavior.

File: src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb [hunk 1]
```diff
@@ -357,7 +357,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
...
 
         Private Function PartOfSingleLineLambda(node As SyntaxNode) As Boolean
-            Return node.AncestorsAndSelf().Any(Function(n) TypeOf n Is SingleLineLambdaExpressionSyntax)
+            While node IsNot Nothing
+                If TypeOf node Is MultiLineLambdaExpressionSyntax Then Return False
+                If TypeOf node Is SingleLineLambdaExpressionSyntax Then Return True
+                node = node.Parent
+            End While
+            Return False
+            'Return node.AncestorsAndSelf().Any(Function(n) TypeOf n Is SingleLineLambdaExpressionSyntax)
```
Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb]: Remove the comment?

Comment (mattwar) [inline src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb]: Damn, you are good!

**Comment by mattwar** [comment]
@Pilchie @mattgertz this one has been waiting long time for ship room?

**Comment by Pilchie** [comment]
My bad, forgot to mark it appropriately.

**Comment by Pilchie** [comment]
:+1:

**Comment by mattwar** [comment]
reissued PR against master as #3268

**Pull request closed without merging**